src/ncurses_ui.cc

```diff
@@ -661,12 +662,55 @@ Optional<Key> NCursesUI::get_next_key()
         const int new_c = wgetch(m_window);
         if (new_c == '[') // potential CSI
         {
-            const Codepoint csi_val = wgetch(m_window);
-            switch (csi_val)
+            const Codepoint c1 = wgetch(m_window);
+            switch (c1)
             {
                 case 'I': return {Key::FocusIn};
                 case 'O': return {Key::FocusOut};
-                default: break; // nothing
+                case '1':
+                {
+                  const Codepoint c2 = wgetch(m_window);
+                  if (c2 != ';')
+                  {
+                      ungetch(c2); ungetch(c1);
+                      break;
+                  }
+
+                  const Codepoint c3 = wgetch(m_window);
+                  function<Key(Key)> f;
+                  switch (c3)
+                  {
+                      case '2': f = shift; break;
+                      case '3': f = alt; break;
+                      case '4': f = shift_alt; break;
+                      case '5': f = ctrl; break;
+                      case '6': f = shift_ctrl; break;
+                      case '7': f = alt_ctrl; break;
+                      case '8': f = shift_alt_ctrl; break;
+                  }
+                  if (f)
```

Comment:
I think this is the wrong way around, you want `if (not f)`

Reply:
Ya, that was stupid, fixed.